numpy/distutils/ccompiler.py

```diff
@@ -19,6 +20,52 @@
                                       quote_args, get_num_build_jobs
 
 
+def newest_file(files):
+    """
+    Return modification time of the newest file in the input
+
+    Parameters
+    ----------
+    file : list
+        Files to check
+
+    Returns
+    -------
+    newer modification time of input
+    """
+    if not files:
+        return 0
+    newest = 0
+    for f in files:
+        newest = max(os.stat(f)[ST_MTIME], newest)
```

Comment:
this returns the newest filename I guess the function should be called newest_file_time